server/src/main/java/io/crate/statistics/ColumnStatsBuilder.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Licensed to Crate.io GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.statistics;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Random;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+
+import io.crate.Streamer;
+import io.crate.data.breaker.RamAccounting;
+import io.crate.types.DataType;
+
+public class ColumnStatsBuilder<T> {
```

Comment:
Will this later contain the sketches? Currently it seems strange that this is streamable too, instead of streaming the result directly.

Maybe worth adding a unit test for this. There are some codecov warnings about missing coverage for the `merge`

Reply:
The followup will actually break this into two parts: the Builder which does the sampling, and then a separate immutable `ColumnSketch<T>` class which will handle serialization and merging.  Given that I'm not sure it's worth adding unit tests yet.

Reply:
Okay, fine to keep as is if it will change anyway before merge to master.